### PR TITLE
DW-4056 - Added Concourse PR job

### PR DESCRIPTION
**Pull request by benj-fletch**

### CODE REVIEW

File: ci/jobs/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+jobs:
+  - name: dataworks-analytical-service-infra-pr
+    plan:
+      - get: dataworks-analytical-service-infra-pr
+        trigger: true
+        version: every
+      - put: dataworks-analytical-service-infra-pr
+        params:
+          context: $BUILD_JOB_NAME
```
Comment (ChrisScottThomas) [inline ci/jobs/pull_request.yml]: You can remove this line, and any other line where this occurs. It's not handled in v6+ currently, and injects a default of 'status'
```suggestion
```

File: ci/resource_types.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+resource_types:
+- name: pull-request
+  type: docker-image
+  source:
+    repository: jtarchie/pr
```
Comment (ChrisScottThomas) [inline ci/resource_types.yml]: We are now using `teliaoss/github-pr-resource` as `jtarchie/pr` is deprecated.  This means some things will have changed when calling this resource. All docker image resource types now require a version as standard.  See: https://github.com/telia-oss/github-pr-resource for reference

```suggestion
    repository: teliaoss/github-pr-resource
    tag: latest
```

File: ci/resources.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+resources:
+  - name: dataworks-analytical-service-infra-pr
+    type: pull-request
+    source:
+      access_token: ((github_com.token))
```
Comment (ChrisScottThomas) [inline ci/resources.yml]:
```suggestion
      access_token: ((dataworks-secrets.concourse_github_pat))
```

File: ci/resources.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+      repo: dwp/dataworks-analytical-service-infra
+      uri: https://github.com/dwp/dataworks-analytical-service-infra.git
```
Comment (ChrisScottThomas) [inline ci/resources.yml]: No longer required.  This is now obtained querying the API and using the repository name.
```suggestion
```

File: ci/resources.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+resources:
+  - name: dataworks-analytical-service-infra-pr
+    type: pull-request
+    source:
+      access_token: ((github_com.token))
+      repo: dwp/dataworks-analytical-service-infra
```
Comment (ChrisScottThomas) [inline ci/resources.yml]:
```suggestion
      repository: dwp/dataworks-analytical-service-infra
```

File: ci/jobs/pull_request.yml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+          platform: linux
+          image_resource:
+            type: docker-image
+            source:
+              repository: ((terraform.repository))
+              version: ((terraform12.version))
+              tag: ((terraform12.version))
+          params:
+            AWS_ACCESS_KEY_ID: ((ci.aws_access_key_id))
+            AWS_SECRET_ACCESS_KEY: ((ci.aws_secret_access_key))
```
Comment (ChrisScottThomas) [inline ci/jobs/pull_request.yml]: Any truly secret, secret should be contained in `dataworks-secrets` this is now: 

```suggestion
            AWS_ACCESS_KEY_ID: ((dataworks-secrets.ci_aws_access_key_id))
            AWS_SECRET_ACCESS_KEY: ((dataworks-secrets.ci_aws_secret_access_key))
```

File: ci/jobs/pull_request.yml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+          DEPLOY_PATH: infra
+        input_mapping:
+          dataworks-analytical-service-infra: dataworks-analytical-service-infra-pr
+      - task: terraform-plan-infra
+        config:
+          platform: linux
+          image_resource:
+            type: docker-image
+            source:
+              repository: ((terraform.repository))
```
Comment (ChrisScottThomas) [inline ci/jobs/pull_request.yml]: This is now a SecretsManager handled parameter and is called as follows.  See https://github.ucds.io/dip/dataworks-secrets for referencing.

```suggestion
              repository: ((dataworks.terraform_repository))
```

File: ci/jobs/pull_request.yml [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+              version: ((terraform12.version))
+              tag: ((terraform12.version))
```
Comment (ChrisScottThomas) [inline ci/jobs/pull_request.yml]: is now `((dataworks.terraform_version))`, as 12 is preferred.  For context TF11 is now, `((dataworks.terraform_11_version))`

```suggestion
              version: ((dataworks.terraform_version))
              tag: ((dataworks.terraform_version))
```

File: ci/resources.yml [hunk 9]
```diff
@@ -0,0 +1,8 @@
...
+      uri: https://github.com/dwp/dataworks-analytical-service-infra.git
+    check_every: 15m
```
Comment (ChrisScottThomas) [inline ci/resources.yml]: These PR checks are super lightweight and can be ran as often as required.

File: ci/meta.yml [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+              cp terraform/deploy/$DEPLOY_PATH/terraform.tf ../terraform-config
+              cp terraform/deploy/$DEPLOY_PATH/terraform.tfvars ../terraform-config
+          dir: dataworks-analytical-service-infra
+        inputs:
+          - name: dataworks-analytical-service-infra
+        outputs:
+          - name: terraform-config
+      params:
+        AWS_ACCESS_KEY_ID: ((ci.aws_access_key_id))
+        AWS_SECRET_ACCESS_KEY: ((ci.aws_secret_access_key))
```
Comment (ChrisScottThomas) [inline ci/meta.yml]: See above

**Review by ChrisScottThomas — CHANGES_REQUESTED** [review]

**Comment by benj-fletch** [comment]
Thanks for the comments @ChrisScottThomas. I have made the relevant changes that you've suggested. I also realised that `aviator.yml` was missed in the initial PR so I've added this.

**Comment by benj-fletch** [comment]
This has now been `aviator`'d and so is ready for review

**Review by ChrisScottThomas — APPROVED** [review]

**Pull request merged**